### ISSUE TITLE
ComposerOrchestrator does not look for composer.phar except on Windows (Could not find a Composer executable.)

### ISSUE THREAD

**Issue by Pimm** [box-project/box#993]
## Bug report

| Question    | Answer
| ------------| ---------------
| Box version | main
| PHP version | all
| Platform with version | all non-Windows

[ComposerOrchestrator](https://github.com/box-project/box/blob/9068269c6268bd48f173b37b6d961b7b26cd4d2b/src/Composer/ComposerOrchestrator.php#L168) currently contains the following code:

```php
$executableFinder = new ExecutableFinder();
$executableFinder->addSuffix('.phar');
$composer = $executableFinder->find('composer');
```

From what I understand, this is supposed to find the Composer executable even if its filename is `composer.phar`, as it is if one has followed [the official download instructions](https://getcomposer.org/download/).

It turns out, [ExecutableFinder ignores suffixes](https://github.com/symfony/symfony/issues/12247) altogether on non-Windows platforms. This is expected behaviour. (Expected by whom? [The documentation](https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Process/ExecutableFinder.php#L35) does not at all hint that `addSuffix` is a no-op outside Windows. But that's a story for another day.)

On my Ubuntu machine, this caused:
```
Could not find a Composer executable.
```
I worked around this issue by creating a `composer` → `composer.phar` symlink.

I'm opening this issue as you might want to consider changing your code to prevent this confusion in the future. I found [this PR](https://github.com/sensiolabs/melody/pull/37) in another project which applied such a change.

**Comment by theofidry** [comment]
Hi, I revisited the issue.

From what I see be it `composer` or `composer.phar`, the binary looked up is first and foremost the one present in the `PATH`.

I am not sure if it's worth changing this logic, I at least didn't see many projects adapting it. However, it is at the very least useful to be able to point Box to a specific binary, maybe via an environment variable.

**Comment by theofidry** [comment]
Actually re-discovered an existing feature, you can specify which Composer executable to use via the `--composer-bin` option of the `compile` command (see #887) available since [4.3.0](https://github.com/box-project/box/releases/tag/4.3.0).
